### DISCUSSION TITLE
Meta param in an assign action is getting the previous state values

### DISCUSSION THREAD

**Discussion by cicerohen** [statelyai/xstate#1309] [General]
I'm try to get a  specific state value insiide a assign action, but  meta param returns the previous state. How can I to make a side effect based on  specific current state values? :/

**Comment by davidkpiano** [comment]
✏️ Please create a [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed) reproduction of this. Thanks!

**Reply by cicerohen** [reply]
Let me explain a little bit better..

I need to execute a specific side-effect function  when fetchCards is on **success** and **filterCards.period** changes. That side-effect function need to know what is the current state of filterCards.period. But,  when the side-effect function is created using assign, the meta param is always getting the previous state. :/   

![Screenshot from 2020-07-16 00 18 26](https://user-images.githubusercontent.com/819041/87622685-05d21880-c6fa-11ea-945e-80c6f37486bd.png)


https://codesandbox.io/s/xstate-example-template-vanilla-ghl6d?file=/src/index.js

**Reply by cicerohen** [reply]
Can you suggest me some alternative approach? :smile:

**Reply by Andarist** [reply]
This is a problematic case - you could try to split your `active` state into 2 and transition from the first one to the second one immediately: https://codesandbox.io/s/xstate-example-template-vanilla-m4r31

This is a hack though, but I'm not sure how to do this differently right now. I would consider this case to be a problem - it clearly is important for `onEntry` to be aware of the entered state and there are SCXML tests for this behavior: https://www.w3.org/Voice/2013/scxml-irp/411/test411.txml

**Reply by cicerohen** [reply]
@Andarist thanks. I will check these links to try figure out a solution for my case. :)

**Reply by cicerohen** [reply]
@Andarist for my case I did this...

I created an assign action to save  the state of filterCards.period to context, then, the someEffect get this value and  make your effect with a real and updated state value...

OK, I know that context is not for this type of data, but...

![Screenshot from 2020-07-16 10 09 03](https://user-images.githubusercontent.com/819041/87674817-ea442d80-c74c-11ea-8dbc-af9709e24d2c.png)
![Screenshot from 2020-07-16 10 08 24](https://user-images.githubusercontent.com/819041/87674843-f4662c00-c74c-11ea-81fa-9dce196f166f.png)
![Screenshot from 2020-07-16 10 15 47](https://user-images.githubusercontent.com/819041/87675157-5e7ed100-c74d-11ea-8f76-ab9395f6a0bd.png)
![Screenshot from 2020-07-16 10 15 34](https://user-images.githubusercontent.com/819041/87675175-63dc1b80-c74d-11ea-8328-eadfa664fd85.png)

![Screenshot from 2020-07-16 10 17 04](https://user-images.githubusercontent.com/819041/87675279-8bcb7f00-c74d-11ea-90d6-8914790f9055.png)
